client/tests/tests.go

```diff
@@ -45,6 +46,23 @@ func (s TestsClient) List(namespace string, tags []string) (*testsAPI.TestList,
 	return toReturn, nil
 }
 
+func (s TestsClient) ListTags(namespace string) ([]string, error) {
+	tags := []string{}
+	list := &testsAPI.TestList{}
+	err := s.Client.List(context.Background(), list, &client.ListOptions{Namespace: namespace})
+	if err != nil {
+		return tags, err
+	}
+
+	for _, test := range list.Items {
+		tags = append(tags, test.Spec.Tags...)
```

Comment:
I would add only new tags there, rather than cleaning duplicates later

Reply:
```
	var (
		a intsets.Sparse
		b bool
	)
	b = a.Insert(9)
	println(b) // true
	b = a.Insert(9)
	println(b) // false
```

Reply:
or simpler 
```
func AppendIfMissing(slice []int, i int) []int {
    for _, ele := range slice {
        if ele == i {
            return slice
        }
    }
    return append(slice, i)
}
```

Reply:
what is your concern here? I can do it but the code will get more complicated and performance wise there is no difference.

Reply:
actually you are right
No need to overdo it